components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/CustomTabWindowFeature.kt

```diff
@@ -57,16 +57,29 @@ class CustomTabWindowFeature(
     }
 
     /**
-     * Starts the feature and a observer to listen for window requests.
+     * Starts observing the configured session to listen for window requests.
      */
     override fun start() {
-        sessionManager.findSessionById(sessionId)?.register(windowObserver)
+        scope = store.flowScoped { flow ->
+            flow.mapNotNull { state -> state.findCustomTab(sessionId) }
+                .ifChanged {
+                    it.content.windowRequest
+                }
+                .collect { state ->
+                    val windowRequest = state.content.windowRequest
+                    if (windowRequest?.type == WindowRequest.Type.OPEN) {
```

Comment:
What would `windowRequest?.type == WindowRequest.Type.CLOSE` mean in this context? Ideally you'd want to close a custom tab, right?

Reply:
Yeah, exactly. If we ever want to support `window.close` for `CustomTab`s we'd have to add that, but we don't support that right now. In fact, we haven't even wired up engine gecko to do that.

Reply:
We might do this soon for https://github.com/mozilla-mobile/android-components/issues/4711 but for now we should be good.

Reply:
:+1: